deeplake/enterprise/dataloader.py

```diff
@@ -785,17 +783,28 @@ def __iter__(self):
 
         dataset_read(self._orig_dataset)
 
-        if self._internal_iterator is not None:
-            self._internal_iterator = iter(self._internal_iterator)
+        if self._iterator is not None:
+            self._iterator = iter(self._iterator)
```

Comment:
Whenever the dataloader is getting intialized better to make the _iterator also initialize in that call.